packages/core/admin/server/controllers/api-token.js

```diff
@@ -53,6 +54,26 @@ module.exports = {
     ctx.created({ data: accessToken });
   },
 
+  async refresh(ctx) {
+    const { id } = ctx.params;
+    const apiTokenService = getService('api-token');
+
+    const apiToken = await apiTokenService.getById(id);
+    if (!apiToken) {
+      ctx.notFound('API Token not found');
+      return;
+    }
+
+    if (!apiToken.lifespan) {
+      ctx.badRequest(`API Token doesn't have a lifespan to refresh`);
```

Comment:
I've mixed feeling about this, I want to use a ValidationError, but I don't know if we can really consider this check as a validation check.